### ISSUE TITLE
Handler processing of new jobs is "unfair"

### ISSUE THREAD

**Issue by hexylena** [galaxyproject/galaxy#11130]
Noticed on EU which currently has 22k 'new' jobs.

![image](https://user-images.githubusercontent.com/458683/104590985-5d7f1b80-566c-11eb-9556-281083a89d2d.png)

Queueing of these jobs is significantly unfair, it's a FIFO queue. This contrasts sharply with our DRM which can enforce user priorities and fairness, and the handlers themselves which have concurrency limits to preserve QoS. One user can still completely overwhelm the queue and prevent anyone else's job from being queued.

As collections grow larger and the ability to launch ridiculous numbers of jobs gets easier, think it might be worth considering making the handler job selection algorithm be more fair than just selecting a batch of N jobs in new state ordered by ID.

**Comment by natefoo** [comment]
Concurrency limits handle the "fairness" in my case but yeah #11358 can be used for this as well.

**Comment by hexylena** [comment]
Yeah, too bad our concurrency limits are 4k which eliminates that benefit we might get.

**Comment by bgruening** [comment]
Just for my understanding, why does concurrent jobs has something to do with this issue? Let's assume we have infinite cluster capacity. 4k concurrent running jobs for one user is ok in this case. But still, the job scheduling takes time and we can still have 27k jobs on the `new` state and users submitting just one job will still be affected. 
In my understanding concurrent-jobs handle fair distribution of `running` job to resources. Nate's patch is handling fiar handling of jobs in `new` state, isn't it? Or do I miss something?
